system/utils.py

```diff
@@ -403,11 +472,21 @@ def check_no_failures(hosts):
 
 async def ensure_primary_changed(pool_handler, wallet_handler, trustee_did, primary_before):
     return await eventually(
-        check_primary_changed, pool_handler, wallet_handler, trustee_did, primary_before,
-        retry_wait=20, timeout=480
+        check_primary_changed, pool_handler, wallet_handler, trustee_did, primary_before, retry_wait=20, timeout=480
     )
 
 
+async def check_pool_availability(pool_handle, wallet_handle, trustee_did):
+    req = await ledger.build_get_validator_info_request(trustee_did)
+    results = json.loads(await ledger.sign_and_submit_request(pool_handle, wallet_handle, trustee_did, req))
+    results = {k: json.loads(v) for k, v in results.items()}
+    assert all([v['result']['data']['Pool_info']['Unreachable_nodes_count'] == 0 for k, v in results.items()])
```

Comment:
Name of this check is a bit misleading in my opinion. Probably `check_all_nodes_connected` or `check_nodes_dont_have_connection_issues` or just `check_all_nodes_online` can be better.

Reply:
Done.